src/core/public/core_app/errors/public_base_url.tsx

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { EuiButton, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+
+import { HttpSetup, NotificationsSetup } from '../..';
+import { DocLinksSetup } from '../../doc_links';
+import { mountReactNode } from '../../utils';
+
+/** Only exported for tests */
+export const MISSING_CONFIG_STORAGE_KEY = `core.warnings.publicBaseUrlMissingDismissed`;
+export const MISMATCHED_CONFIG_STORAGE_KEY = `core.warnings.publicBaseUrlMismatchedDismissed`;
+
+/** Small helper component for the toast content */
+const WarningToastContent: React.FC<{ docLinks: DocLinksSetup; onMute: () => void }> = ({
+  children,
+  docLinks,
+  onMute,
+}) => {
+  return (
+    <>
+      <p>
+        {children}
+        <a href={`${docLinks.links.settings}#server-publicBaseUrl`} target="_blank">
```

Comment:
NIT: From your screenshots, there is not space between the warning message and the link.